### PR TITLE
install pre-commit

### LINKED ISSUE #42: install pre-commit
## install pre-commit :
- install pre-commit . [click here](https://www.npmjs.com/package/pre-commit).
- make configuration pre-commit [here](https://github.com/observing/pre-commit)

### PR DESCRIPTION
**Pull request by khamis-hesham**
fix #42
- [x] install pre-commit . [click here](https://www.npmjs.com/package/pre-commit).
- [x] make configuration pre-commit [here](https://github.com/observing/pre-commit)
- [x] make lint-fix script for check the code before loading

### CODE REVIEW

**Review by imansedky5 — DISMISSED** [review]

**Review by yasmeenattallah — DISMISSED** [review]

File: package.json [hunk 1]
```diff
@@ -10,8 +10,15 @@
     "test": "NODE_ENV=test jest --runInBand --detectOpenHandles --coverage",
     "kill-run": "npm run kill && npm run dev",
     "start": "NODE_ENV=production node ./server/index.js",
-    "heroku-postbuild": "NODE_ENV=production cd client && npm install && npm run build"
+    "heroku-postbuild": "NODE_ENV=production cd client && npm install && npm run build",
+    "lint-fix": "eslint server/ --fix",
+    "lint": "eslint server/",
+    "lint-react": "cd client && npm run lint"
```
Comment (mFlaifel) [inline package.json:16]: please add react lint fix to the script;
```json
"lint-react-fix": "cd client && npm run lint-fix"
```

File: package.json [hunk 2]
```diff
@@ -10,8 +10,15 @@
...
   },
+  "pre-commit": [
+    "test",
+    "lint-fix"
+  ],
```
Comment (mFlaifel) [inline package.json:21]: and add lint react to precommit,
```suggestion
 "pre-commit": [
    "test",
    "lint-fix",
    "lint-react-fix"
  ],
```

**Review by mFlaifel — CHANGES_REQUESTED** [review]
good work

**Review by AdhamHaisami — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (khamis-hesham) [reply package.json:16]: Ok

File: package.json (hunk 2 above)
Comment (khamis-hesham) [reply package.json:21]: Done

File: client/package.json [hunk 3]
```diff
@@ -41,6 +43,6 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
-  }, 
+  },
   "proxy": "http://localhost:5000"
-}
+}
```
Comment (hassanelnajjar) [inline client/package.json:48]: Hello Khamis, you need to add line here :) 
I think you can fix this by changing the default formater in .vscode
Please add this to .vscode setting 
 "editor.defaultFormatter": "dbaeumer.vscode-eslint",

**Review by mFlaifel — APPROVED** [review]

File: client/package.json (hunk 3 above)
Comment (khamis-hesham) [reply client/package.json:48]: Thx Hasan :heart:

**Pull request merged**